### PR TITLE
Fix #4494: Hint bulb animation added

### LINKED ISSUE #4494: Make the hint button animate.
**Is your feature request related to a problem? Please describe.**
From researcher feedback, students do not realize when a hint is available (or when a new hint is available that's different from one that they've already seen). They've also seemed fairly reticent to click on the hint button.

**Describe the solution you'd like**
Make the light bulb on the Hints button bounce up and down (possibly with some pause in between) when a hint is available. It doesn't have to be too overt; it just needs to 'softly' get the user's attention.

### PR DESCRIPTION
**Pull request by rt4914**
## Explanation
Fixes #4494

This PR introduces a bouncing animation for the light bulb to make it easier to notice. It has an animation delay of 5 seconds and will repeat every 30 seconds. The animation itself lasts about 6 seconds (see the video below). The animation will continue until the user reveals the hint or moves on to the next card.

## Essential Checklist

- [x] The PR title and explanation each start with "Fix #bugnum: " (If this PR fixes part of an issue, prefix the title with "Fix part of #bugnum: ...".)
- [x] Any changes to [scripts/assets](https://github.com/oppia/oppia-android/tree/develop/scripts/assets) files have their rationale included in the PR explanation.
- [x] The PR follows the [style guide](https://github.com/oppia/oppia-android/wiki/Coding-style-guide).
- [x] The PR does not contain any unnecessary code changes from Android Studio ([reference](https://github.com/oppia/oppia-android/wiki/Guidance-on-submitting-a-PR#undo-unnecessary-changes)).
- [x] The PR is made from a branch that's **not** called "develop" and is up-to-date with "develop".
- [x] The PR is **assigned** to the appropriate reviewers ([reference](https://github.com/oppia/oppia-android/wiki/Guidance-on-submitting-a-PR#clarification-regarding-assignees-and-reviewers-section)).

## For UI-specific PRs only
Demonstration video:

https://user-images.githubusercontent.com/12983742/185574373-27abd959-a377-49b5-a677-56a799d9ee1b.mp4

### CODE REVIEW

**Comment by rt4914** [comment]
In this PR Bazel related CI checks are failing which I am unable to solve fully right now.
1. I think I need to mention the newly added `hint_bulb_animation.xml` file somewhere in bazel files for correct build but I am unable to find the correct file.
2. There is some CI check because of TODO check but I am finding it weird as why this check is failing because it seems it should not be related to this PR.

**Comment by rt4914** [comment]
> In this PR Bazel related CI checks are failing which I am unable to solve fully right now.
> 
> 1. I think I need to mention the newly added `hint_bulb_animation.xml` file somewhere in bazel files for correct build but I am unable to find the correct file.
> 2. There is some CI check because of TODO check but I am finding it weird as why this check is failing because it seems it should not be related to this PR.

All CI checks are passing now.

File: app/src/main/java/org/oppia/android/app/player/state/StateFragmentPresenter.kt [hunk 1]
```diff
@@ -459,31 +462,58 @@ class StateFragmentPresenter @Inject constructor(
...
+      val hintBulbAnimation = AnimationUtils.loadAnimation(
+        context,
+        R.anim.hint_bulb_animation
+      )
+      binding.hintBulb.startAnimation(
+        hintBulbAnimation
+      )
+
+      // The hint bulb will blink 5 times in duration of 6 sec, stop for next 24 sec and again
+      // start repeating.
```
Comment (BenHenning) [inline app/src/main/java/org/oppia/android/app/player/state/StateFragmentPresenter.kt]: Should this be repeated every 30 seconds? The current implementation will only repeat it once.

File: app/src/main/java/org/oppia/android/app/player/state/StateFragmentPresenter.kt [hunk 2]
```diff
@@ -459,31 +462,58 @@ class StateFragmentPresenter @Inject constructor(
...
   }
+
+  private fun setHintOpenedAndUnRevealed(isHintRevealed: Boolean) {
+    viewModel.setHintOpenedAndUnRevealedVisibility(isHintRevealed)
+    if (isHintRevealed) {
+      val hintBulbAnimation = AnimationUtils.loadAnimation(
+        context,
+        R.anim.hint_bulb_animation
+      )
+      binding.hintBulb.startAnimation(
```
Comment (BenHenning) [inline app/src/main/java/org/oppia/android/app/player/state/StateFragmentPresenter.kt]: For here & below: will the animation be correctly stopped if the hint state changes?

File: app/src/main/java/org/oppia/android/app/player/state/StateFragmentPresenter.kt [hunk 3]
```diff
@@ -338,7 +341,7 @@ class StateFragmentPresenter @Inject constructor(
           oppiaLogger.e("StateFragment", "Failed to retrieve hint/solution", result.error)
         } else {
           // If the hint/solution, was revealed remove dot and radar.
-          viewModel.setHintOpenedAndUnRevealedVisibility(false)
+          setHintOpenedAndUnRevealed(false)
```
Comment (BenHenning) [inline app/src/main/java/org/oppia/android/app/player/state/StateFragmentPresenter.kt:346]: Reminders to add tests & add this to questions' implementation, too.

File: app/src/main/res/anim/hint_bulb_animation.xml [hunk 4]
```diff
@@ -0,0 +1,12 @@
+<?xml version="1.0" encoding="utf-8"?>
+<set xmlns:android="http://schemas.android.com/apk/res/android"
+  android:fillAfter="true"
+  android:interpolator="@android:anim/bounce_interpolator">
+  <alpha
+    android:duration="600"
+    android:fromAlpha="0.0"
+    android:interpolator="@android:anim/accelerate_interpolator"
```
Comment (BenHenning) [inline app/src/main/res/anim/hint_bulb_animation.xml]: Why an accelerator interpolator? To have a bit of a "pulse"?

**Review by BenHenning — COMMENTED** [review]
Thanks @rt4914! The video seems a bit like the animation is intermittent, but it might just be the recording and/or my computer. I'll check it out locally to see how it looks on a device.

**Comment by seanlip** [comment]
Just a note -- agreed with @BenHenning that the pulsing animation looks a bit weird. It looked better in the [old video](https://user-images.githubusercontent.com/9396084/184732142-588e49a3-5ded-41bd-8ccd-9db69691815c.mp4).

The behaviour of "flash 5 seconds, stop 25 seconds" seems fine to me though! Thanks Rajat :)

File: app/src/main/res/anim/hint_bulb_animation.xml (hunk 4 above)
Comment (BenHenning) [reply app/src/main/res/anim/hint_bulb_animation.xml]: I've changed the interpolation logic from the original solution, so this comment is no longer relevant.

File: app/src/main/java/org/oppia/android/app/player/state/StateFragmentPresenter.kt (hunk 2 above)
Comment (BenHenning) [reply app/src/main/java/org/oppia/android/app/player/state/StateFragmentPresenter.kt]: Verified that it does work correctly.

File: app/src/main/java/org/oppia/android/app/player/state/StateFragmentPresenter.kt (hunk 1 above)
Comment (BenHenning) [reply app/src/main/java/org/oppia/android/app/player/state/StateFragmentPresenter.kt]: Verified the original solution didn't repeat, but the changes do.

**Review by BenHenning — APPROVED** [review]
Approving this with some open comments. #4510 is tracking addressing those comments post-merge. Plus, an additional review should take place for the changes I contributed directly to the PR (since they haven't yet been reviewed).

**Pull request merged**